Otus-DevOps-2020-02/DmitryKorlas_microservices
# Monitoring 2

# Completed the homework №21


## What were done:
 - detach monitoring-related things into docker-compose-monitoring.yml
 - connect cAdvisor monitoring service
 - add grafana visualisation tool
 - add grafana dashboard
 - add grafana dashboard UI_Service_Monitoring
 - add grafana dashboard Business_Logic_Monitoring
 - add alertmanager service and setup alert to notify when post service is down

## How to run the project:
 ```sh
  cd docker 
  cp .env.example .env 
  docker-compose -f docker-compose.yml -f docker-compose-monitoring.yml config
```

## How to check, that it works correctly:
 Visit following URL's to see the services:  
 - http://34.78.251.183 - our service
 - http://34.78.251.183:9090 - prometheus control panel
 - http://34.78.251.183:9090/alerts - should contains "InstanceDown" alert
 - http://34.78.251.183:3000 - grafana control panel
 - http://34.78.251.183:9093 - alertmanager control panel
 
 check alert sub-system: 
 - stop post service `docker-compose stop post`
 - wait for 1m
 - check slack - it should contains a notification message *[FIRING:1] InstanceDown (post:5000 post page)*
 - check alertmanager, it should display InstanceDown record
 - check alerts in prometheus, it should display FIRE on **InstanceDown** alert
 - start post service back `docker-compose start post`
 - check alerts in prometheus, **InstanceDown** should be inactive

## PR checklist:
 - [x] Added a label with the number of homework
